// @apollo/server/src/externalTypes/index.ts
/**
 * This file represents the external type definitions needed by end users,
 * integration implementors, and plugin implementors. Everything from this file
 * is re-exported by the root (via * export), so add exports to this file with
 * intention (it's public API).
 */
export type { BaseContext, ContextFunction, ContextThunk } from '@apollo/server/src/externalTypes/context';
export type { GraphQLRequest, GraphQLResponse } from '@apollo/server/src/externalTypes/graphql';
export type {
  HTTPGraphQLRequest,
  HTTPGraphQLResponse,
  HTTPGraphQLHead,
} from '@apollo/server/src/externalTypes/http';
export type {
  ApolloServerPlugin,
  GraphQLFieldResolverParams,
  GraphQLRequestExecutionListener,
  GraphQLRequestListener,
  GraphQLRequestListenerDidResolveField,
  GraphQLRequestListenerExecutionDidEnd,
  GraphQLRequestListenerParsingDidEnd,
  GraphQLRequestListenerValidationDidEnd,
  GraphQLSchemaContext,
  GraphQLServerListener,
  GraphQLServerContext,
  LandingPage,
} from '@apollo/server/src/externalTypes/plugins';
export type {
  GraphQLRequestContext,
  GraphQLRequestMetrics,
  GraphQLRequestContextDidEncounterErrors,
  GraphQLRequestContextDidResolveOperation,
  GraphQLRequestContextDidResolveSource,
  GraphQLRequestContextExecutionDidStart,
  GraphQLRequestContextParsingDidStart,
  GraphQLRequestContextResponseForOperation,
  GraphQLRequestContextValidationDidStart,
  GraphQLRequestContextWillSendResponse,
} from '@apollo/server/src/externalTypes/requestPipeline';
export type {
  DocumentStore,
  ApolloConfigInput,
  ApolloConfig,
  PersistedQueryOptions,
  CSRFPreventionOptions,
  ApolloServerOptionsWithSchema,
  ApolloServerOptionsWithTypeDefs,
  ApolloServerOptionsWithStaticSchema,
  ApolloServerOptionsWithGateway,
  ApolloServerOptions,
} from '@apollo/server/src/externalTypes/constructor';

export type {
  GraphQLExperimentalFormattedInitialIncrementalExecutionResult,
  GraphQLExperimentalFormattedSubsequentIncrementalExecutionResult,
  GraphQLExperimentalFormattedIncrementalResult,
  GraphQLExperimentalFormattedIncrementalDeferResult,
  GraphQLExperimentalFormattedIncrementalStreamResult,
} from '@apollo/server/src/externalTypes/incrementalDeliveryPolyfill';
